gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -0,0 +1,396 @@
+fluid.onUncaughtException.addListener(function (err) {
+    gpii.eventLog.logError(null, "UncaughtException", err);
+}, "eventLog");
+
+// Log fluid.fail.
+fluid.failureEvent.addListener(function (args) {
+    var err = Array.isArray(args) ? args.join(" ") : err;
+    gpii.eventLog.logError(null, "Failure", err);
+}, "eventLog", "before:fail");
+
+/**
+ * Gets the path of the new log file for this instance of gpii.
+ *
+ * @return {String} Path of the log file.
+ */
+gpii.eventLog.getLogFilePath = function () {
+    if (!gpii.eventLog.logFilePath) {
+        var startupTime = Date.now();
+        gpii.eventLog.logFilePath = process.env.GPII_EVENT_LOG;
+
+        if (!gpii.eventLog.logFilePath) {
+            var settingsDirComponent = gpii.settingsDir();
+            var gpiiSettingsDir = settingsDirComponent.getGpiiSettingsDir();
+            gpii.eventLog.logFilePath =
+                path.join(gpiiSettingsDir, "gpii-" + gpii.journal.formatTimestamp(startupTime) + ".log");
+        }
+
+        fluid.log(fluid.logLevel.IMPORTANT, "Writing event log to " + gpii.eventLog.logFilePath);
+    }
+    return gpii.eventLog.logFilePath;
+};
+
+/**
+ * Writes an event to the log file.
+ *
+ * @param level {Object} Level of the log, see fluid.logLevelsSpec [FATAL,FAIL,WARN,IMPORTANT,INFO,TRACE].
+ * @param event {Object} The object.
+ */
+gpii.eventLog.writeLog = function (level, event) {
+    var intLevel = gpii.eventLog.checkLevel(level);
+    event.level = intLevel.value;
+
+    // Log to console before the installation ID and timestamp are added (no one wants to see it).
+    fluid.log(intLevel, event);
+
+    if (!gpii.eventLog.logFilePath) {
+        gpii.eventLog.logFilePath = gpii.eventLog.getLogFilePath();
+    }
+
+    event.installID = gpii.eventLog.installationID;
+    event.timestamp = gpii.eventLog.getTimestamp();
+    fs.appendFileSync(gpii.eventLog.logFilePath, JSON.stringify(event) + "\n");
+};
+
+/**
+ * Starts a timer.
+ *
+ * @param timerId {String} Something to identify the timer.
+ */
+gpii.eventLog.startTimer = function (timerId) {
+    // hrtime used instead of Date.now to ignore time changes.
+    gpii.eventLog.timers[timerId] = process.hrtime();
+};
+
+/**
+ * Stops a timer (removes it from the collection), and returns the time in seconds since it started.
+ *
+ * @param timerId {String} Something to identify the timer.
+ * @return {Number} Number of seconds since the start of the timer.
+ */
+gpii.eventLog.stopTimer = function (timerId) {
+    var seconds = 0;
+    var start = gpii.eventLog.timers[timerId];
+    if (start) {
+        var duration = process.hrtime(start);
+        // Round to the nearest second.
+        seconds = duration[0] + (duration[1] < 5e8 ? 0 : 1);
+    }
+    return seconds;
+};
+
+/**
+ * Determines if an object is empty (it has no properties of its own)
+ *
+ * @param obj The object to check.
+ * @return {boolean} true for {}, null, or undefined.
+ */
+gpii.eventLog.isEmpty = function (obj) {
```

Comment:
We supply $.isEmpty from jQuery which you can include via var $ = fluid.registerNamespace("jQuery") which I assume this is a near-duplicate of